code401-challenges/src/main/java/code401/challenges/linkedlist/LinkedList.java

```diff
@@ -135,13 +135,51 @@ public int insertAfter(int value, int newVal){
                     newValue.next = current.next;
                     current.next = newValue;
                     this.size++;
-                    return 0;
+                    return;
                 }else{
                     current = current.next;
                 }
             }
         }
         System.out.println("The linked list does not contain "+ value);
+        return;
+    }
+
+    public int knthFromEnd(int value){
+        if(this.head == null){
+            throw new NullPointerException("The Linked list is empty");
+        }else if(value > this.size){
+            throw new IllegalArgumentException("The value you passed is greater than the size of the linked list. Please try a different value");
+        }else if (value == this.size){
+               throw new IllegalArgumentException("The value you passed is the same as the size of the linked list");
+        }else if(value < 0) {
+            throw new IllegalArgumentException("You passed in a negative value");
+        }else{
+            //the kIndex represents the difference between the size of the linked list and the k index. I.E it represents the index of the value we are going to be returning
+            int kIndex = this.size - value;
+            int listCounter = 1;
+            Node current = this.head;
+
+            while(current != null){
+                if(kIndex == listCounter){
+                    return current.value;
+                }else{
+                    current = current.next;
+                    listCounter++;
+                }
+            }
+        }
         return -1;
```

Comment:
return `-1` here is a bad idea. The user won't know if `-1` is the value k-the-from-the-end or if `-1` means that the value wasn't found